gpkit/nomials/nomial_math.py

```diff
@@ -633,14 +633,27 @@ def as_posyslt1(self):
         "Returns the posys <= 1 representation of this constraint."
         s = self._unsubbed.sub(self.substitutions, require_positive=False)
         posy, negy = s.posy_negy()
-        if len(negy.cs) != 1:
-            raise ValueError("SignomialInequality could not simplify to"
-                             " a PosynomialInequality")
-        else:
+        oper = "<="
+        if negy is 0:
+            raise ValueError("SignomialConstraint %s became the infeasible"
+                             " constraint %s %s %s after substitution." %
+                             (self, posy, oper, negy))
+        elif posy is 0:
+            raise ValueError("SignomialConstraint %s became the tautological"
+                             " constraint %s %s %s after substitution." %
+                             (self, posy, oper, negy))
```

Comment:
should the order of the two cases above be swapped, so that `0 <= 0` returns the tautological error instead of infeasible?